onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -171,6 +171,90 @@ activation.)DOC")
       })
       .SetDoc(R"DOC(Tokenizer divides each string in X into a vector of strings along the last axis. All input strings including attributes are UTF-8 encoded.)DOC");
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(Ngram)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Input(0, "X", "Input for n-gram extraction", "T")
+      .Output(0, "Y", "NGram results", "T1")
+      .TypeConstraint(
+          "T",
+          {"tensor(string)", "tensor(int32)", "tensor(int64)"},
+          "Input is ether string UTF-8 or int32/int64")
+      .TypeConstraint(
+          "T1",
+          {"tensor(float)"},
+          "1-D tensor of floats")
+      .Attr(
```

Comment:
What's the reason for using float as the output type?

Reply:
They are counts and possibly scaled by inverse document frequency.